daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ChoiceGroup.scala

```diff
@@ -74,12 +75,12 @@ trait ChoiceDefMixin extends AnnotatedSchemaComponent with GroupDefLike {
     new DFDLChoice(newDFDLAnnotationXML("choice"), this)
 
   lazy val xmlChildren = xml match {
-    case <choice>{c @ _*}</choice> => c
-    case <group>{_*}</group> => {
+    case c if c.label == "choice" => c.nonEmptyChildren
```

Comment:
Hmmm. I expected this to become pattern matching on the Elem class:
```
case Elem(_, "choice", _, _, c) => c
case ELem(_,"group", _, _, g) => .....
```
The args to Elem are (prefix, label, attributes, namespaceBindings, children).
Pattern matching like this is semantically identical to the XML destructuring.  
I.e., 'c' is bound to a Seq[Node] just as in the XML destructuring. 
Your solution has changed the meaning of 'c' to be the parent of those children.